### PR TITLE
bluetooth: fast_pair: Rename Fast Pair sample

### PR DESCRIPTION
**Pull request by alstrzebonski**
Created fast_pair folder in samples folder and renamed peripheral_fast_pair sample to input_device sample. Changed the device name in Google Nearby Devices console.

Jira: NCSDK-22586

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -332,11 +332,12 @@ Bluetooth samples
 
   * Fixed an interoperability issue with iOS devices by setting the report IDs of HID input and output reports to zero.
 
-* :ref:`peripheral_fast_pair` sample:
+* :ref:`fast_pair_input_device` sample:
```
Comment (kapi-no) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:335]: you may need to align the name for the past release notes (release notes 2.0.0, 2.1.0, 2.2.0, 2.4.0)

File: samples/bluetooth/fast_pair/input_device/sample.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 sample:
-  name: Fast Pair Provider
-  description: Bluetooth Fast Pair Provider sample
+  name: Fast Pair Provider input device
```
Comment (kapi-no) [inline samples/bluetooth/fast_pair/input_device/sample.yaml:2]: I would use the same name as is defined in the sample Readme file.

```suggestion
  name: Fast Pair Input Device
```

File: samples/bluetooth/fast_pair/input_device/sample.yaml [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+  description: Bluetooth Fast Pair Provider input device sample
```
Comment (kapi-no) [inline samples/bluetooth/fast_pair/input_device/sample.yaml:3]:
```suggestion
  description: Bluetooth Fast Pair Provider sample setup for the Input Device use case 
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -332,11 +332,12 @@ Bluetooth samples
...
 
   * Added automatic switching to the Fast Pair not discoverable advertising mode with the hide UI indication instead of removing the Fast Pair advertising payload when all bond slots are taken.
   * Updated by disabling the :kconfig:option:`CONFIG_BT_SETTINGS_CCC_LAZY_LOADING` Kconfig option as a workaround fix for the `Zephyr issue #61033`_.
   * Fixed an issue where the sample was unable to advertise in Fast Pair not discoverable advertising mode when it had five Account Keys written.
+  * Renamed the sample to :ref:`fast_pair_input_device` and relocated it to the :file:`samples/fast_pair` folder.
```
Comment (kapi-no) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:340]:
```suggestion
  * Renamed the sample to :ref:`fast_pair_input_device` and relocated it to the :file:`samples/bluetooth/fast_pair` folder.
```

**Review by kapi-no — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (alstrzebonski) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:335]: Ok, I was wondering if I should do it or not.

**Comment by alstrzebonski** [comment]
Addressed review comments.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (kapi-no) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:335]: I am not sure if we shouldn't use the label value here (that is "Bluetooth: Fast Pair"). @divipillai ?

File: doc/nrf/releases_and_maturity/releases/release-notes-2.0.0.rst [hunk 5]
```diff
@@ -294,7 +294,7 @@ Bluetooth samples
 * Added:
 
   * :ref:`peripheral_ams_client` sample.
-  * :ref:`peripheral_fast_pair` sample.
+  * ``peripheral_fast_pair`` sample.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-2.0.0.rst:297]:
```suggestion
  * Bluetooth: Fast Pair sample.
```
Same in other places also.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (divipillai) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:335]: We provide the sample name in old release notes. I have added it as a suggestion for clarity.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -332,11 +332,12 @@ Bluetooth samples
 
   * Fixed an interoperability issue with iOS devices by setting the report IDs of HID input and output reports to zero.
 
-* :ref:`peripheral_fast_pair` sample:
+* :ref:`fast_pair_input_device` sample:
 
   * Added automatic switching to the Fast Pair not discoverable advertising mode with the hide UI indication instead of removing the Fast Pair advertising payload when all bond slots are taken.
   * Updated by disabling the :kconfig:option:`CONFIG_BT_SETTINGS_CCC_LAZY_LOADING` Kconfig option as a workaround fix for the `Zephyr issue #61033`_.
   * Fixed an issue where the sample was unable to advertise in Fast Pair not discoverable advertising mode when it had five Account Keys written.
+  * Renamed the sample to :ref:`fast_pair_input_device` and relocated it to the :file:`samples/bluetooth/fast_pair` folder.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:340]:
```suggestion
  * Renamed the sample to :ref:`fast_pair_input_device` and moved it to the :file:`samples/bluetooth/fast_pair` folder.
```

**Comment by peknis** [comment]
The doc build error is: /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/samples/bluetooth/fast_pair/input_device/README.rst: WARNING: document isn't included in any toctree

Hot tip: Search for "nrf/src" in the error log to find the culprit.

**Comment by divipillai** [comment]
> The doc build error is: /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/samples/bluetooth/fast_pair/input_device/README.rst: WARNING: document isn't included in any toctree
> 
> Hot tip: Search for "nrf/src" in the error log to find the culprit.

yeah, we should add "   ../../../samples/bluetooth/fast_pair/*/README" in the bl.rst file.

File: samples/bluetooth/fast_pair/input_device/README.rst [hunk 7]
```diff
@@ -1,13 +1,13 @@
-.. _peripheral_fast_pair:
+.. _fast_pair_input_device:
 
-Bluetooth: Fast Pair
-####################
+Bluetooth: Fast Pair Input Device
+#################################
```
Comment (divipillai) [inline samples/bluetooth/fast_pair/input_device/README.rst:4]:
```suggestion
Bluetooth: Fast Pair input device
#################################
```
Also confirmed with @peknis

File: samples/bluetooth/fast_pair/input_device/README.rst [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 
 .. contents::
    :local:
    :depth: 2
 
-The Fast Pair sample demonstrates :ref:`how to use Google Fast Pair with the nRF Connect SDK <ug_bt_fast_pair>`.
+The Fast Pair Input Device sample demonstrates :ref:`how to use Google Fast Pair with the nRF Connect SDK <ug_bt_fast_pair>`.
```
Comment (divipillai) [inline samples/bluetooth/fast_pair/input_device/README.rst:10]:
```suggestion
The Fast Pair input device sample demonstrates :ref:`how to use Google Fast Pair with the nRF Connect SDK <ug_bt_fast_pair>`.
```

**Comment by alstrzebonski** [comment]
Addressed review comments. Please rereview.

**Review by mkapala-nordic — APPROVED** [review]

**Comment by alstrzebonski** [comment]
Rebased on main.

**Review by divipillai — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

File: doc/nrf/samples/bl.rst [hunk 9]
```diff
@@ -9,3 +9,4 @@ Bluetooth samples
    :glob:
 
    ../../../samples/bluetooth/*/README
+   ../../../samples/bluetooth/fast_pair/*/README
```
Comment (kapi-no) [inline doc/nrf/samples/bl.rst:12]: is it possible to sort the displayed sample list alphabetically? It seems that the Fast Pair sample is currently at the very bottom of the list

Comment (divipillai) [reply doc/nrf/samples/bl.rst:12]: yeah, I can check this ,and can be improved later.

**Comment by alstrzebonski** [comment]
Rebased on main.

**Pull request merged**